trin-core/src/types/receipts.rs

```diff
@@ -276,42 +345,72 @@ mod tests {
-        let transactions = Receipts {
+        let receipts = Receipts {
             receipt_list: vec![
-                Receipt::decode(&hex::decode(RECEIPT_0).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_1).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_2).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_3).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_4).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_5).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_6).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_7).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_8).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_9).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_10).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_11).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_12).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_13).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_14).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_15).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_16).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_17).unwrap()).unwrap(),
-                Receipt::decode(&hex::decode(RECEIPT_18).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_0).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_1).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_2).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_3).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_4).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_5).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_6).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_7).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_8).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_9).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_10).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_11).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_12).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_13).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_14).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_15).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_16).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_17).unwrap()).unwrap(),
+                rlp::decode(&hex::decode(RECEIPT_18).unwrap()).unwrap(),
```

Comment:
I guess I have the same issue here, with `rlp` not feeling right. If I'm not making sense, then maybe we can have a call?